feat: add `blas/ext/index-of` (#7565)
stdlib-js/stdlib: lib/node_modules/@stdlib/blas/ext/index-of/docs/repl.txt

        shape defined by `options.dims`. For example, given the input shape
        `[2, 3, 4]` and `options.dims=[0]`, the search element ndarray must have
        a shape which is broadcast-compatible with the shape `[3, 4]`. Similarly
        when performing the operation over all elements in a provided input
        ndarray, a search element ndarray must be a zero-dimensional ndarray.

    fromIndex: ndarray|integer (optional)
        Index from which to begin searching. May be either a scalar value or an
        ndarray having an `integer` or `generic` data type. If provided an
        ndarray the value must have a shape which is broadcast compatible with
        the complement of the shape defined by `options.dims`. For example,
        given the input shape `[2, 3, 4]` and `options.dims=[0]`, an ndarray
        containing the index from which to begin searching must have a shape
        which is broadcast-compatible with the shape `[3, 4]`. Similarly
        when performing the operation over all elements in a provided input
        ndarray, an ndarray containing the index from which to begin searching

>         must be a zero-dimensional ndarray. By default, the value of the index
>         from which to begin searching is `0`.

Author: ```suggestion
        must be a zero-dimensional ndarray. By default, the index
        from which to begin searching is `0`.
```

    options: Object (optional)
        Function options.

    options.dtype: string (optional)

>         Output array data type. Must be a "integer_index_and_generic" data type.

Author: ```suggestion
        Output array data type. Must be an integer or generic data type.
```
